deploy/test-environments/main.tf

```diff
@@ -54,25 +56,26 @@ provider "restapi" {
 
 # Elastic Cloud (EC) deployment
 module "ec_deployment" {
-  count  = var.serverless_mode ? 0 : 1
-  source = "github.com/elastic/apm-server/testing/infra/terraform/modules/ec_deployment"
+  count = var.serverless_mode ? 0 : 1
 
+  source        = "../cloud/modules/ec"
+  ec_api_key    = var.ec_api_key
   region        = var.ess_region
-  stack_version = var.stack_version
+  stack_version = local.cleaned_version
```

Comment:
So we don't even deploy the requested version? Where do we use the commit sha?

Reply:
In the case of a released or snapshot version, `stack_version` will not be modified and remains the same. However, in the case of a BC version, we need to remove the hash from `stack_version`. This is essential because the `ec` provider might fail with an unknown version if the hash is included. Additionally, we need to override `docker_image_tag`. Instead of creating complex logic to handle different scenarios, we always provide `docker_image_override` in all cases to ensure that our deployment deploys the pinned version.